native/src/layout/flex.rs

```diff
@@ -90,16 +116,37 @@ where
         .fill_factor();
 
         if fill_factor == 0 {
-            let (max_width, max_height) = axis.pack(available, max_cross);
+            let (min_width, min_height) = axis.pack(
+                0.0,
+                if align_items == Align::Fill {
+                    cross
+                } else {
+                    0.0
+                },
+            );
 
-            let child_limits =
-                Limits::new(Size::ZERO, Size::new(max_width, max_height));
+            let (max_width, max_height) = axis.pack(
+                available,
+                if align_items == Align::Fill {
+                    cross
+                } else {
+                    max_cross
+                },
+            );
```

Comment:
I wonder if we could group the branching here with:

```rust
let (min_height, max_height) = if align_items == Align::Fill {
    // ...
} else {
   // ...
};
```

It may help with readability!

Reply:
That cleaned it up a bunch, thanks!